src/main/java/Duke/TaskList.java

```diff
@@ -0,0 +1,193 @@
+package Duke;
+import java.util.ArrayList;
+
+public class TaskList {
+    ArrayList<Task> list;
+    int curSize;
+    ArrayList<String[]> data;
+
+
+    public TaskList(ArrayList<String[]> data){
+        this.list = new ArrayList<>();
+        this.curSize = 0;
+        try {
+            this.LoadTask(data);
+        } catch (InputNotValidError e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void LoadTask(ArrayList<String[]> data) throws InputNotValidError{
+        for (String[] each : data){
+            String[] actions = new String[2];
+            String actionName = each[0];
+            Boolean done = true;
+            if(each[1].equals(" 0 ")){
+                done = false;
+            }
+            if (actionName.equals("T ")){
+                actions[0] = "todo";
+                actions[1] = each[2].substring(1, each[2].length());
+            }else if(actionName.equals("E ")){
+                actions[0] = "event";
+                actions[1] = each[2].substring(1) + "/at"  + each[3].substring(1, each[3].length());
+            }else if(actionName.equals("D ")){
+                actions[0] = "deadline";
+                actions[1] = each[2].substring(1) + "/by" + each[3].substring(1, each[3].length());                
+            }
+            Parser hitoryData = new Parser(actions[0] + " " + actions[1]);
+            this.actionHalder(hitoryData, done, true);
+        }
```

Comment:
Perhaps you might want to consider following camelCase standard in the naming of your methods too.